modules/core/shared/src/main/scala/weaver/Comparison.scala

```diff
@@ -0,0 +1,34 @@
+package weaver
+
+import cats.Eq
+import cats.Show
+import com.eed3si9n.expecty._
+
+trait Comparison[A] {
+  def diff(expected: A, found: A): Option[String]
```

Comment:
This result type could be more explicit, if we wanted:
```scala
sealed trait Result
object Result {
   case object Success extends Result
   case object Failure(diff: String) extends Result
}
```

Reply:
I think that's a good idea. There should however be a couple builders in the companion object, for convenience : 

```
def ???[A](f : (A, A) => Option[String]) : Comparison[A] 
// and maybe : 
def ???[A](f: PartialFunction[(A, A), String]) : Comparison[A] 
```

Reply:
I went with `instance` to align with [Eq](http://typelevel.org/cats/api/cats/kernel/Eq$.html#instance[A](f:(A,A)=%3EBoolean):cats.kernel.Eq[A]).